### DISCUSSION TITLE
How to define a certain start and end time for the waveform that is being generated?

### DISCUSSION THREAD

**Discussion by Ahmed-Ateeq** [katspaugh/wavesurfer.js#3845] [Q&A]
Hello, I was hoping to generate waveforms with variable start and end time. My audios can be trimmed from either the start or the end meaning that the start and end time can change, and along with that I want to potentially show the waveform in a dynamic way. What may be the best way to achieve this using wavesurfer.js?

**Comment by katspaugh** [comment]
There's no built-in way to do it. Wavesurfer will render the entire audio you pass. You can trim it in your app and do `wavesurfer.loadBlob`, see https://wavesurfer.xyz/docs/classes/wavesurfer.default#loadBlob
